certutils/reloader.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code is based on the example in https://github.com/kelseyhightower/kube-cert-manager/blob/master/tls-app/certificate-manager.go
+
+package certutils
+
+import (
+	"crypto/tls"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// CertReloader provides a mechanism for reloading a TLS key and cert upon file change
+type CertReloader struct {
+	sync.RWMutex
+	certificate *tls.Certificate
+	certFile    string
+	keyFile     string
+	Error       chan error
+	watcher     *fsnotify.Watcher
+}
+
+// NewCertReloader returns a new CertReloader
+func NewCertReloader(certFile, keyFile string) (*CertReloader, error) {
+	cr := &CertReloader{
+		certFile: certFile,
+		keyFile:  keyFile,
+		Error:    make(chan error, 10),
+	}
+	err := cr.setCertificate()
+	if err != nil {
+		return nil, err
+	}
+
+	go cr.watchCertificate()
```

Comment:
Should we wrap `cr.watchCertificate()` with a func to catch its error?

Reply:
sounds reasonable